ipld/retriever.go

```diff
@@ -50,19 +52,66 @@ func NewRetriever(dag format.DAGService, codec rsmt2d.Codec) *Retriever {
 func (r *Retriever) Retrieve(ctx context.Context, dah *da.DataAvailabilityHeader) (*rsmt2d.ExtendedDataSquare, error) {
 	ses := r.newSession(ctx, dah)
 	for _, qs := range newQuadrants(dah) {
-		// TODO: ErrByzantineRow/Col
 		for _, q := range qs {
 			eds, err := ses.retrieve(ctx, q)
 			if err == nil {
 				return eds, nil
 			}
+			if byzErr := r.checkForByzantineError(ctx, dah, err); byzErr != nil {
+				return nil, byzErr
+			}
```

Comment:
After we update to the new rsmt2d version, we can make just one if here for `ErrByzantineData` and extract the logic into `handleByzantineError`